src/zkevm_specs/evm/main.py

```diff
@@ -15,24 +15,26 @@ def verify_steps(
     begin_with_first_step: bool = False,
     end_with_last_step: bool = False,
 ):
-    # TODO: Enforce general ExecutionState transition constraint
+    # For the last step, the next step is meaningless
+    if end_with_last_step:
+        steps += [None]
+
+    for idx in range(len(steps) - 1):
+        curr, next = steps[idx], steps[idx + 1]
```

Comment:
We could do something like
```suggestion
    for curr, _next in zip(steps, steps[1:]):
```
This works even if `steps` has 1 element.

Reply:
I just found we still need the `idx`, so I keep the current version. Also in #116, we can't extend the `steps` with a `None`, so iterating on range seems easier.